cmake/linker/lld/target_base.cmake

```diff
@@ -11,6 +11,7 @@ macro(toolchain_ld_base)
   # TOOLCHAIN_LD_FLAGS comes from compiler/clang/target.cmake
   # LINKERFLAGPREFIX comes from linker/lld/target.cmake
   zephyr_ld_options(
+    -no-pie
```

Comment:
Do you think other linkers will need this also?  Maybe it should be added to compiler/clang/target.cmake instead.

Reply:
I am not sure. I basically copied it from https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/linker/ld/target_base.cmake#L14 to the same place for lld. But i think it would probably be good to align setting the "-no-pie" option for the compiler and linker in a similar way.

Reply:
this looks reasonable to me with the current macro based linker handling we have.

At some point we might we might want to align more to the compiler approach, as in #24851, but that's a different story.